messagingpush/src/main/java/io/customer/messagingpush/data/communication/CustomerIOPushNotificationCallback.kt

```diff
@@ -11,27 +10,27 @@ interface CustomerIOPushNotificationCallback {
      *
      * @param context reference to application context
      * @param payload data received for the notification
-     * @return [TaskStackBuilder] to launch activities on notification click;
-     * null to let the SDK handle this
+     * @return [Unit] non null, depicting that customer is going to launch activities on notification click;
+     * and null to let the SDK handle this
      *
      * NOTE: If your app is targeting Android 12 or greater, be careful of
      * launching intents outside the app as it can affect the notification
      * open metrics tracking
      */
-    fun createTaskStackFromPayload(
-        context: Context,
-        payload: CustomerIOParsedPushPayload
-    ): TaskStackBuilder? = null
+    fun onNotificationClicked(
+        payload: CustomerIOParsedPushPayload,
+        context: Context
+    ): Unit? = null
```

Comment:
If I'm thinking right, we're here trying to simplify so if customers just implement the function, it means we should not process further. To stop any further action, they should return `null` explicitly. I like the idea though, but I think this could be confusing and error-prone. Can we stick to simple `boolean` instead for simplification? 🤔 

Might not be very good example though, but [`onInterceptTouchEvent`](https://developer.android.com/reference/android/view/ViewGroup#onInterceptTouchEvent(android.view.MotionEvent)) and many others use `boolean` for similar cases. And I think we should stick with boolean too 😄

Reply:
I don't mind it being `boolean`, but i feel like its confusing. `onInterceptTouchEvent` true or false both provide a different meaning. In our case, it means, if user want to handle the `notification clicked` they are going to provide us with action to perform. 

Otherwise, we do what we do. They would only need to provide `null` if they first up, start overriding the optional callback on purpose and then wants to tell us, to still not cater for what they provided 😄

Reply:
If you say so, I'm okay with either option 😄